manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts

```diff
@@ -89,6 +89,7 @@ module Apiman {
                 $scope.updateButton.state = 'in-progress';
                 var gateway = Gateway();
                 ApimanSvcs.update({ entityType: 'gateways', secondaryType: $scope.gateway.id }, gateway, function() {
+                    $rootScope.isDirty = false;
```

Comment:
@kahboom - this was the fix to the edit-gateway page.  The bug was that I was getting the "you haven't saved your changes yet!" dialog when clicking on the 'Save' button.  The reason is that the 'Save' button on that page results in a redirect of the user to a new page.  So the solution was to set the isDirty flag to false if the save worked, prior to doing the redirect. :)